core/libcamera_app.cpp

```diff
@@ -381,7 +381,10 @@ void LibcameraApp::ConfigureVideo(unsigned int flags)
 	// Now we get to override any of the default settings from the options_->
 	StreamConfiguration &cfg = configuration_->at(0);
 	cfg.pixelFormat = libcamera::formats::YUV420;
-	cfg.bufferCount = 6; // 6 buffers is better than 4
+	if (options_->buffer_count > 0)
+	{
+		cfg.bufferCount = options_->buffer_count;
+	}
```

Comment:
The reasoning behind this change is reasonable.  However, we are working to integrate something like this in the core of libcamera.  You can see the patches [here](https://patchwork.libcamera.org/project/libcamera/list/?series=3555).  I would prefer if we did not add similar code to libcamera-apps.

Reply:
Given that this change is not related to AF, would you be able to create a new PR for this?  We can then merge it straight away.

Reply:
I think we should probably add this change to all the `ConfigureXxx` methods? That way it's a consistent option across the board.

Reply:
Though having said that... maybe the viewfinder mode wants a separate `--viewfinder-buffer-count` option, because you might want a different number from the still option. In fact you certainly would.

Reply:
I will remove this from this PR and add the suggested --viewfinder-buffer-count